optimade/server/config.py

```diff
@@ -157,6 +157,12 @@ class ServerConfig(BaseSettings):
         None, description="Host settings to pass through to the `Elasticsearch` class."
     )
 
+    mongo_count_timeout: int = Field(
+        5,
+        description="""Number of seconds to allow MongoDB to perform a full database count before falling back to `null`.
+This operation can require a full COLLSCAN for empty queries which can be prohibitively slow if the database does not fit into the active set, hence a timeout can drastically speed-up response times.""",
```

Comment:
You are right, it shouldn't be slow, but previously for an empty filter we were just naively calling `count_documents` which does still do a full scan that can be very slow. Now, I am using `estimated_document_count` for this case, which uses simple collection metadata to just return the number. However for filters, it can still make use of this timeout.